experiments/react-native-uifabric/src/components/Persona/Persona.types.ts

```diff
@@ -0,0 +1,46 @@
+import { IPersonaCoinProps, IPersonaCoinTokens } from '../PersonaCoin';
+import { ViewProps, TextProps } from 'react-native';
+import { IRenderData } from '@uifabricshared/foundation-composable';
+import { ITextTokens } from '../../../src/tokens/TextTokens';
+
+export const personaName = 'RNFPersona';
+
+export interface IPersonaState {
+  text?: string;

```

Comment:
The state here is actually the existence of text so these should be bools, right? Though maybe we don't have a good way to retain the text values outside of `renderData.state`. Is it possible to merge the text into the slot directly, it's a child, which is just another prop. You're probably doing it the best way expressible right now, but I don't know how I feel about keeping text strings in state.

Reply:
The reason I do this is so that I can omit the slot altogether if the text is empty. 
` {!!text && <Slots.text>{text}</Slots.text>}`
If it insert the text directly into the slot, the slot is always rendered.

Reply:
I think the actual problem I have is that the word `state` means so many different things and I kind of wish renderData split things into state and info or something and that state actually reflected state in the `react` state and info got used for the things we need to keep around for rendering.

Reply:
@dotnetjunky yup, I understand 😊. I just have a problem with the word state being used everywhere. I think what you have is totally reasonable, it would be cool though if you the text value was merged into the slot directly and all you needed to do was keep a bool around to tell you whether to render that slot. But that's a problem for another time

Reply:
I agree. `state` here should have been called `data` or something like that.